_source/_posts/2022-01-19-react-storybook.md

```diff
@@ -0,0 +1,688 @@
+
+```js
+import { useEffect, useState } from 'react';
+import { useOktaAuth } from '@okta/okta-react';
+
+export const useAuth = () => {
+  const { oktaAuth, authState } = useOktaAuth();
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    if (authState?.isAuthenticated) {
+      if (!user) {
+        oktaAuth.getUser().then(setUser);
+      }
+    } else {
+      setUser(null);
+    }
+  }, [authState, user, oktaAuth]);
+
+  const login = async () => oktaAuth.signInWithRedirect('/');
+  const logout = async () => oktaAuth.signOut('/');
+
+  return [user, login, logout];
+};
+```
+
+Finally, you need to modify the existing `App` component to use the authentication hook. Open `src/App.js` and adjust the content to look like this.
+
+```js
+import './App.css';
+import { Link } from 'react-router-dom';
+import { Header } from './stories/Header';
+
+export const App = ({useAuth}) => {
+  const [user, login, logout] = useAuth();
+
+  return (
+    <div className="App">
+      <Header user={user} onLogin={login} onLogout={logout} />
+      <h1>Unit Converter</h1>
+      <p>
+      <Link to="/converter">Go to the app!</Link>
+      </p>
+    </div>
+  );
+}
+```
+
+Congratulations, you have completed your React application with Storybook. You can now open the console in the project folder and run the following command to start the app.
+
+```bash
+npm start
+```
+
+You should see the application's front page in your browser. When you click the  **Go to the app!** link, you'll log in on the Okta page. After successfully signing in, you will redirect to the unit converter page, which looks like the image below.
+
+{% img blog/react-storybook/react-storybook-application.jpg alt:"The finished unit conversion application" width:"800" %}{: .center-image }
+
+## Learn more about React, Storybook, and Single-Page Apps
+
+In this tutorial, I have shown you how to create a React application and use Storybook to browse the application's components. Storybook is a great tool that can enhance your development workflow.
+* It lets you view and test your components in isolation. 
+* You can specify the location of each component in a hierarchical menu and then browse through the components in your browser. 
+* You can have multiple stories showcasing it in different use-cases for each component. 
```

Comment:
```suggestion
* You can have multiple stories showcasing it in different use cases for each component. 
```
removed hyphen. 
unclear to me what the "it" refers to here, could you say "showcasing  different use cases for each component" or does that lose the point? Thx.

Reply:
Yep, I think components makes sense here I'll make the change